packages/api/src/router/user.ts

```diff
@@ -81,7 +90,16 @@ export const userRouter = createTRPCRouter({
         .from(users)
         .where(eq(users.id, input.userId))
         .limit(1);
+      const existingUser = await ctx.db.query.users.findFirst({
+        where: eq(users.name, input.form.name),
+      });
 
+      if (existingUser !== undefined) {
```

Comment:
I would not show an error when you rename to your current username, you can simply check if the existing users id is not the updated users id

Reply:
It's not there that I checked, it's below in the condition that makes sure the form is dirty before it allows pressing the save button (i.e. the values are not the default ones anymore)